src/components/RegionSelectInput/index.tsx

```diff
@@ -0,0 +1,57 @@
+import type { Props as SelectInputProps } from '#components/SelectInput';
+import SelectInput from '#components/SelectInput';
+import { useRequest } from '#utils/restRequest';
+import { numericIdSelector } from '#utils/selectors';
+
+function regionNameSelector({ region_name }: { region_name: string }) {
+    return region_name;
+}
+
+interface RegionOption {
+    id: number;
+    region_name: string;
+}
+
+type Props<NAME> = SelectInputProps<
+    number,
+    NAME,
+    RegionOption,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    any,
+    'value' | 'name' | 'options' | 'keySelector' | 'labelSelector'
+> & {
+    className?: string;
+    name: NAME;
+    onChange: (newValue: number | undefined, name: NAME) => void;
+    value: number | undefined | null;
+}
+
+function RegionSelectInput<NAME>(props: Props<NAME>) {
+    const {
+        className,
+        name,
+        onChange,
+        value,
+        ...otherProps
+    } = props;
+
+    const { response: regionResponse } = useRequest({
+        url: '/api/v2/region/',
+    });
```

Comment:
This select input should be lazy.
Let's be careful so that we do not send 100x requests because someone used this in a list.